### ISSUE TITLE
Include option to replace spaces for special characters

### ISSUE THREAD

**Issue by 8ryo** [iafisher/obsidian-quick-links#6]
Some search links use '+', some use '_', some use '%20'. An extra field for the user to enter in a custom space character(s) would be great.

**Comment by 8ryo** [comment]
This is what I mean btw, dunno if it breaks anything though (I've never used typescript before). For main.ts

```
export interface QuickLinkMacro {
  prefix: string;
  target: string;
  space_character: string;
}

export const DEFAULT_QUICK_LINKS: QuickLinkMacro[] = [
  // Wikipedia
  {
    prefix: "w",
    target:
      "https://www.wikipedia.org/w/index.php?title=Special:Search&search=%s",
    space_character: "_" // Added space character
  },
  // Blank entry for the settings editor
  {
    prefix: "",
    target: "",
    space_character: "" // Added space character
  },
];

export interface RawLink {
  text: string;
  target: string;
  em: boolean,
}

export function transformLink(
  link: RawLink,
  quickLinksMap: Map<string, QuickLinkMacro>
): RawLink | null {
  const linkPrefix = getLinkPrefix(link.target);
  if (linkPrefix === "") {
    return null;
  }

  const quickLink = quickLinksMap.get(linkPrefix);
  if (quickLink === undefined) {
    return null;
  }

  const linkHrefNoPrefix = link.target.slice(quickLink.prefix.length + 1).replace(/ /g, space_character); //Added space character change
  const displayText =
    // [](w:Whatever) --> linkText === ""
    // [[w:Whatever]] --> linkHref === linkText
    //
    // In either case, we should use "Whatever" as the display text.
    //
    // Otherwise, the user has set some custom display text and we should
    // use that.
    link.text === "" || link.target === link.text
      ? linkHrefNoPrefix
      : link.text;

  const linkTarget = quickLink.target.replace("%s", linkHrefNoPrefix);
  return { target: linkTarget, text: displayText, em: link.em };
}

export function getLinkPrefix(linkHref: string): string {
  const linkSplit = linkHref.split(":", 2);
  if (linkSplit.length !== 2) {
    return "";
  }
  return linkSplit[0];
}
```

**Comment by iafisher** [comment]
Hi, thanks for the suggestion. Do you have an example of a  search URL that needs a custom space character, so I can test it?

**Comment by 8ryo** [comment]
@iafisher 
> Hi, thanks for the suggestion. Do you have an example of a search URL that needs a custom space character, so I can test it?

https://en.wikipedia.org/wiki/%s

This one requires uses underscore (i.e. '_')  for spaces, thanks.

**Comment by iafisher** [comment]
Thanks, this is implemented in v0.2.4, which I just released.

**Comment by 8ryo** [comment]
> Thanks, this is implemented in v0.2.4, which I just released.

Awesome! Much appreciated!
